modules/javafx.graphics/src/main/java/com/sun/javafx/css/StyleManager.java

```diff
@@ -767,25 +768,24 @@ Image getCachedImage(String url) {
                         if (image.isError()) {
                             final PlatformLogger logger = getLogger();
                             if (logger != null && logger.isLoggable(Level.WARNING)) {
-                                logger.warning("Error loading image: " + url);
+                                DataURI dataUri = DataURI.tryParse(url);
+                                if (dataUri != null) {
+                                    logger.warning("Error loading image: " + dataUri);
+                                } else {
+                                    logger.warning("Error loading image: " + url);
+                                }
```

Comment:
So if I understand this correctly, an Image is first loaded via the standard mechanism, and if that fails for any reason (`image.isError()`) a second attempt is made through `DataURI` ?

Would it not be better to register a new protocol so `new Image` would just succeed for data uri's ?

Reply:
In this specific case, image loading has failed for some reason. The call to `DataURI.tryParse` is only there to potentially call `DataURI.toString()` for a truncated log output, instead of logging the entire `url` String. If data-URI truncation is not wanted, this code can be reverted.

Reply:
Maybe add a brief comment to that effect?

Reply:
Ok clear thanks

Reply:
I added a comment that explains that.